File: M/485-max-consecutive-ones/main.go
package main

func findMaxConsecutiveOnes(nums []int) int {
	c := 0
	max := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] == 1 {
			c++
		} else {
			if c > max {
				max = c
			}
			c = 0
		}
	}
	if c > max {
		max = c
	}
	return max
}

func main() {

}

File: M/485-max-consecutive-ones/main.rs
struct Solution {}

impl Solution {
  pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
    let mut cnt = 0;
    let mut max = 0;
    for i in nums {
      if i == 1 {
        cnt += 1;
      } else {
        if max < cnt {
          max = cnt;
        }
        cnt = 0;
      }
    }
    if max < cnt {
      max = cnt;
    }
    max
  }
}

fn main() {
  println!("{}", Solution::find_max_consecutive_ones(vec![0; 5]));
}
